utils.go

```diff
@@ -16,3 +18,23 @@ func TransformEach[T any](items []T, fn func(T) Node) []Node {
 	}
 	return nodes
 }
+
+// EscapeCommentContents escapes the contents of a comment node to ensure safe rendering according to https://html.spec.whatwg.org/multipage/syntax.html#comments
+func EscapeCommentContents(s string) string {
+	// escape disallowed sequences
+	s = strings.ReplaceAll(s, "<!--", "&lt;!--")
```

Comment:
Each call is O(N) and allocates a new string. You can collapse all of your substitutions into a single `strings.Replacer` (one pass, fewer allocations).

```go
var commentReplacer = strings.NewReplacer(...)

// Then in the function
s := commentReplacer.Replace(...)
```